File: lib/year2019/day05.rb
require 'solver'
require 'year2019/intcode_computer'

module Year2019
  class Day05 < Solver
    def part_one
      IntcodeComputer.run(program, inputs: [1]).output
    end

    def part_two
      IntcodeComputer.run(program, inputs: [5]).output
    end

    def program
      @program ||= raw_input.chomp.split(",").map(&:to_i)
    end
  end
end

File: rust/src/year2019/day05.rs
use crate::io::{Input, Solution};
use crate::year2019::computer::Program;

pub fn solve(part: &str, input: &Input) -> Solution {
    Solution::build(part, input, &part_one, &part_two)
}

fn part_one(input: &Input) -> i64 {
    let mut program = Program::from((input, 1));
    program.run();
    program.output().unwrap()
}

fn part_two(input: &Input) -> i64 {
    let mut program = Program::from((input, 5));
    program.run();
    program.output().unwrap()
}
